homeassistant/components/modem_callerid/__init__.py

```diff
@@ -1 +1,55 @@
-"""The modem_callerid component."""
+"""The Modem Caller ID integration."""
+import logging
+
+from phone_modem import PhoneModem, exceptions
+import serial
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DATA_KEY_API, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [SENSOR_DOMAIN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Modem Caller ID from a config entry."""
+    device = entry.data[CONF_DEVICE]
+    try:
+        api = PhoneModem(device)
+        await api.initialize(device)
+    except (
+        FileNotFoundError,
+        exceptions.SerialError,
+        serial.SerialException,
+        serial.serialutil.SerialException,
```

Comment:
In the future it would be nice to give them a different human readable error on FileNotFound vs serial error but since you have ` from ex` on the `ConfigEntryNotReady` it will already show a different error in the UI.